store/postgres/src/relational/index.rs

```diff
@@ -180,6 +180,22 @@ impl Expr {
         }
     }
 
+    fn is_same_kind_columns(current: &Vec<Expr>, orig: &Vec<Expr>) -> bool {
```

Comment:
Is this checking if `current` and `orig` are the same expressions? I am not entirely clear on that. It feels a little weird that the comparison is then done on the string representation of the `Expr`. It would seem more logical to me if this was a method on `Expr`; since that implements `PartialEq`, would that be the right comparison so that you could just say `current == orig`? If that's not the right comparison, maybe having a method `Expr::is_same(&self, other: &Expr) -> bool` would be clearer.

In that case, this method would boil down to `current.iter().zip(orig.iter()).all(|(c,o)| c.is_same(o))` (I think)

I would also call this method `same_columns` - that name seems a bit clearer.

Reply:
It's not exactly a comparison for equality but is it "same kind" of expression. For example the `Vid` and `Unknown("vid int8_minmax_multi_ops")` are considered "same kind", also `BlockRangeLower` and `Unknown("lower(block_range) int4_minmax_multi_ops)"`. The current implementation checks if the string representation of each expression of the original is a prefix of the current one. Hence the operation is not commutative either. For sure e comment of the functionality would help here so I am adding the summary of this paragraph.

I guess a proper parsing of `Vid`, `BlockRangeLower`, etc. that would keep the optional operator class as a part of the enum, and then compare the enum types and ignore the presence/absence of the operator class would be more robust, and then also `PartialEq` would work, but I took a shortcut that I believe is good enough here. What do you think?

Reply:
Oh .. I see. Yes, that's the downside of this homegrown parsing that it fails for some of the newer things we do because the code was never updated. We should definitely update the code in a follow-on PR and can then hopefully reduce this method to an equality comparison. Or maybe it's time to replace the homegrown code with `sqlparser`

Reply:
I agree that updating the parsing would be really good. Even if we don't end up with equality comparison, it would be much cleaner than the current comparison against a prefix of a string...

Regarding the homegrown code vs sqlparser I was initially all for improving the homegrown code and it kind of works with some minor changes, but now I am slightly in favour for a sqlparser. Implementing the test code made me stumble upon a few edge cases (need for a space after the indexing method, fixed spaces, trailing semicolon, and support for quotes in fieldnames and namespace), while all those are fixable too, I feel that we would benefit from more robust solution than a regex expression.

Reply:
Created an [issue](https://github.com/graphprotocol/graph-node/issues/5512) for this discussion.